CraftTweaker2-API/src/main/java/crafttweaker/api/event/ProjectileImpactArrowEvent.java

```diff
@@ -0,0 +1,48 @@
+package crafttweaker.api.event;
+
+import crafttweaker.annotations.ZenRegister;
+import crafttweaker.api.entity.IEntity;
+import crafttweaker.api.entity.IEntityLivingBase;
+import crafttweaker.api.item.IItemStack;
+import crafttweaker.api.world.IRayTraceResult;
+import stanhebben.zenscript.annotations.ZenClass;
+import stanhebben.zenscript.annotations.ZenGetter;
+import stanhebben.zenscript.annotations.ZenMethod;
+import stanhebben.zenscript.annotations.ZenSetter;
+
+@ZenClass("crafttweaker.event.ProjectileImpactArrowEvent")
+@ZenRegister
+public interface ProjectileImpactArrowEvent extends ProjectileImpactEvent, IEventCancelable {
+  @ZenGetter("arrow")
+  IEntity getArrow();
+
+  @ZenGetter("shooter")
+  IEntity getShooter();
+
+  @ZenGetter("damage")
+  double getDamage();
+
+  @ZenSetter("damage")
+  void setDamage (double damage);
+
+  @ZenSetter("knockbackStrength")
+  void setKnockbackStrength (int knockbackStrength);
+
+  @ZenGetter("isCritical")
+  boolean getIsCritical ();
+
+  @ZenSetter("isCritical")
+  void setIsCritical(boolean isCritical);
+
+  @ZenGetter("pickupStatus")
+  String getPickupStatus();
```

Comment:
For IEventHasResult, you used three methods to check the different status' and three methods to set them.
Here you use one string to get them, and three methods to set them.

Reply:
Perhaps IEventHasResult it could do with a default String implementation that returns the type instead of relying on 3 variables -- one of the things I wasn't sure about with the implementation was to how to specifically handle getting it.

Reply:
I've pushed a change to make IEventHasResult match this and other methods.